code/go/0chain.net/blobbercore/handler/helper.go

```diff
@@ -15,20 +18,57 @@ import (
 	"0chain.net/blobbercore/constants"
 )
 
-func setupGRPCHandlerContext(ctx context.Context, r *blobbergrpc.RequestContext) context.Context {
-	ctx = context.WithValue(ctx, constants.CLIENT_CONTEXT_KEY,
-		r.Client)
-	ctx = context.WithValue(ctx, constants.CLIENT_KEY_CONTEXT_KEY,
-		r.ClientKey)
-	ctx = context.WithValue(ctx, constants.ALLOCATION_CONTEXT_KEY,
-		r.Allocation)
-	return ctx
+// setupGRPCHandlerContext will add client context keys in request context.Context
+// and return new context with client and client key values
+func setupGRPCHandlerContext(ctx context.Context, _allocation string) (context.Context, string, string) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if ok || md == nil {
+		md = metadata.MD{}
+	}
+	getValue := func(key string) string {
+		list := md.Get(key)
+		if len(list) > 0 {
+			return list[0]
+		}
+		return ""
+	}
+
+	client := getValue(common.ClientHeader)
+	clientKey := getValue(common.ClientKeyHeader)
+	// adding to context for future reference and calls
+	ctx = context.WithValue(ctx, constants.CLIENT_CONTEXT_KEY, client)
+	ctx = context.WithValue(ctx, constants.CLIENT_KEY_CONTEXT_KEY, clientKey)
+	ctx = context.WithValue(ctx, constants.ALLOCATION_CONTEXT_KEY, _allocation)
+	return ctx, client, clientKey
+}
```

Comment:
The two times that `setupGRPCHandlerContext` is used, the last parameter, `clientKey`, is not used. If it is unnessary, it should be removed.